src/Engine/Waypoint/Waypoints.hpp

```diff
@@ -186,19 +186,19 @@ class Waypoints {
   }
 
   /**
-   * Generate takeoff waypoint
+   * Generate a temorary waypoint with a given name.
    *
    * @return waypoint copy
    */
-  Waypoint GenerateTakeoffPoint(const GeoPoint &location,
-                                double terrain_alt) const noexcept;
+  Waypoint GenerateTempPoint(const GeoPoint &location, double terrain_alt,
+                             const TCHAR *misc) const noexcept;
```

Comment:
What does the name "misc" mean?

Reply:
I used the the same var name from [InputEvents::eventAddWaypoint(const TCHAR *misc)](https://github.com/XCSoar/XCSoar/pull/1065/files#diff-db7439a8b995f3f0a1251a530565dee3741b42a6413d0b264d5073a735a1b076R651) for consistency, but we can easily change it to something more meaningful (like "name")

Reply:
That's a generic API signature, and the name has no special meaning there, other than "it's some string that the user specified in the file". The function you designed is not a generic API, but a very specific function. It's a different context. Saying you just copied the name is a rather lame excuse for not coming up with a meaningful name (or with API documentation).

Reply:
sure a _lame excuse_... any other preferred name, or `name` is fine?